src/software/thunderscope/proto_unix_io.py

```diff
@@ -82,7 +86,7 @@ def __send_proto_to_observers(self, receive_buffer: ThreadSafeBuffer):
                 except queue.Full:
                     print("Buffer registered to receive everything dropped data")
 
-    def register_observer(self, proto_class, buffer):
+    def register_observer(self, proto_class: Any, buffer: ThreadSafeBuffer) -> None:
```

Comment:
I think for `proto_class` we should use the protobuf [Message](https://googleapis.dev/python/protobuf/latest/google/protobuf/message.html#google.protobuf.message.Message) class as the type annotation. Same goes for any other functions where a protobuf is a param

```suggestion
    def register_observer(self, proto_class: Type[Message], buffer: ThreadSafeBuffer) -> None:
```